app/waf/src/controllers/cert.controller.ts

```diff
@@ -0,0 +1,270 @@
+  patch,
+  del,
+  requestBody,
+  RequestContext,
+  RestBindings,
+  HttpErrors,
+} from '@loopback/rest';
+import {Cert, Adc} from '../models';
+import {CertRepository, AdcRepository} from '../repositories';
+import {Schema, Response, CollectionResponse} from '.';
+import {BaseController} from './base.controller';
+import {inject} from '@loopback/core';
+import {ASGService, ASGManager} from '../services';
+import path = require('path');
+
+const prefix = '/adcaas/v1';
+export class ContentBody {
+  name: string;
+  content: string;
+}
+
+export class CertController extends BaseController {
+  asgMgr: ASGManager;
+
+  constructor(
+    @repository(CertRepository)
+    public certRepository: CertRepository,
+    @repository(AdcRepository)
+    public adcRepository: AdcRepository,
+    @inject(RestBindings.Http.CONTEXT, {optional: true})
+    protected reqCxt: RequestContext,
+    @inject('services.ASGService')
+    public asgService: ASGService,
+  ) {
+    super(reqCxt);
+    this.asgMgr = new ASGManager(this.asgService);
+  }
+
+  @post(prefix + '/certs', {
+    responses: {
+      '200': Schema.response(Cert, 'Successfully create Cert resource'),
+      '400': Schema.badRequest('Invalid Cert resource'),
+      '422': Schema.unprocessableEntity('Unprocessable Cert resource'),
+    },
+  })
+  async create(
+    @requestBody(
+      Schema.createRequest(Cert, 'Cert resource that need to be created'),
+    )
+    cert: Partial<Cert>,
+  ): Promise<Response> {
+    cert.tenantId = await this.tenantId;
+    return new Response(Cert, await this.certRepository.create(cert));
+  }
+
+  @get(prefix + '/certs/count', {
+    responses: {
+      '200': {
+        description: 'Cert model count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async count(
+    @param.query.object('where', getWhereSchemaFor(Cert)) where?: Where,
+  ): Promise<Count> {
+    return await this.certRepository.count(where);
+  }
+
+  @get(prefix + '/certs', {
+    responses: {
+      '200': Schema.collectionResponse(
+        Cert,
+        'Successfully retrieve Cert resources',
+      ),
+    },
+  })
+  async find(
+    @param.query.object('filter', getFilterSchemaFor(Cert))
+    filter?: Filter,
+  ): Promise<CollectionResponse> {
+    return new CollectionResponse(
+      Cert,
+      await this.certRepository.find({
+        where: {or: [{tenantId: await this.tenantId}, {public: true}]},
+      }),
+    );
+  }
+
+  @get(prefix + '/certs/{id}', {
+    responses: {
+      '200': Schema.response(Cert, 'Successfully retrieve Cert resource'),
+      '404': Schema.notFound('Can not find Cert resource'),
+    },
+  })
+  async findById(
+    @param(Schema.pathParameter('id', 'Cert resource ID')) id: string,
+  ): Promise<Response> {
+    return new Response(
+      Cert,
+      await this.certRepository.findById(id, {
+        where: {
+          and: [
+            {or: [{tenantId: await this.tenantId}, {public: true}]},
```

Comment:
I am not quite sure if a 'public' server cert is a valid use case. Maybe we can think about that later.